test_runner/regress/test_sharding_service.py

```diff
@@ -938,3 +938,67 @@ def storage_controller_consistent():
         env.storage_controller.consistency_check()
 
     wait_until(10, 1, storage_controller_consistent)
+
+
+def test_sharding_service_re_attach(neon_env_builder: NeonEnvBuilder):
```

Comment:
nit: kinda feels like you could've extended the heartbeating test above instead of adding a new one, but perhaps a separate one is better

Reply:
The thing I wanted to isolate on this test is the way we have one primary and one secondary on each node, migrated "just so" for the test rather than with default scheduling -- should hopefully be more maintainable than threading that into the heartbeats test.